OpenAI.SDK/OpenAiOptions.cs

```diff
@@ -21,16 +29,34 @@ public class OpenAiOptions
         ///     from an environment variable or key management service.
         /// </summary>
         public string ApiKey { get; set; } = null!;
-
         public string ApiVersion { get; set; } = "v1";
-
         public string BaseDomain { get; set; } = "https://api.openai.com/";
+        public string DeploymentId { get; set; }
 
         /// <summary>
         ///     Default engine id. If you are working with only one engine, this will save you from few line extra code.
         /// </summary>
         public static string? DefaultEngineId { get; set; }
-
+        /// <summary>
+        ///     Create an instance of this class with the necessary information to connect to the azure open ai api
+        /// </summary>
+        /// <param name="baseDomain">Domain for your open ai instance</param>
+        /// <param name="deploymentId">The id of your deployment of open ai</param>
+        /// <param name="apiVersion">The azure open ai api version</param>
+        /// <param name="apiKey">Token used for authentication</param>
+        /// <returns>A valid OpenAiSettings instance configured with the method inputs</returns>
+        static OpenAiOptions CreateAzureSettings(string baseDomain, string deploymentId, string apiVersion, string apiKey)
```

Comment:
Why we need this?

Reply:
It isn't needed so much as it is convenient for people who don't use the appsettings based service collection extensions.  It's hard to know which things are actually required for azure vs not azure with everything being in the same settings class.  I can remove this and change the validation instead though, probably fits with the project better.